### ISSUE TITLE
Machine ID is not available

### ISSUE THREAD

**Issue by drfoland** [bitgapp/Cursr#27]
When I open Cursr in Windows 10, I am unable to view my Machine ID. The Settings > License popup instead shows "Default string" in the field. If I click the copy button, it just copies the text "Default string" to the clipboard. This problem persists even when running it as an administrator. See image below:

![image](https://user-images.githubusercontent.com/21369713/197801510-5cf7124c-4a46-4847-8d73-12e669bff19e.png)

I have successfully activated Cursr before on another computer, also running Windows 10. I have already purchased a second license for this machine, but am unable to activate it due to the fact that I can't generate a Machine ID. Please advise.

**Comment by ragauskl** [comment]
@drfoland yeah it happened before, for now try using 'Default string' as Machine ID (in cursr.app webpage you can manually generate a license and input machine ID), that's how Cursr sees it and all it does is compares the ID to the one Window gives in this case it's 'Default string'.

 It seems like it's a common issue with some hardware where a serial number is not assigned to it on purpose or by accident.

I already have task in my 'to-do' list to find a better solution for Windows instead of depending on hardware to have a serial number, but it's a rare issue (for now at least) and there's a way to make it work so it's of low priority, after next feature release (v1.5.0) I'll be looking into this issue again. When it will be fixed you'll likely see that your license is invalid because of 'Machine Mismatch' or similar and you can then simply revoke and regenerate the license with new machine ID

If solution mentioned above does not work, let me know

**Comment by ragauskl** [comment]
@drfoland This issue has now been fixed, previous license will keep working if already applied and you should not need to do anything, future licenses where possible (if auto license management enabled) will update automatically and if you'll want to apply a license manually you'll need to create new license using the Machine ID in v1.4.12 (if you'll want to reapply old license it will no longer work, so you'll need to revoke it and create new license in that case)
